seed/static/seed/js/controllers/data_quality_admin_controller.js

```diff
@@ -352,6 +353,33 @@ angular.module('BE.seed.controller.data_quality_admin', [])
           show_configuration_errors(misconfigured_rules);
         }
 
+        // Find duplicate rules and trigger warnings
+        var group_by_criteria = function(rules, criteria) {
+          let groups = _.groupBy(rules, function (rule) {
+            return rule[criteria];
+          });
+          var grouped = [];
+          _.forEach(groups, function(group) {
+            if (group.length > 1) grouped.push(group);
+          });
+          return grouped;
+        };
+        var last_rules = [];
+        var recurse = function(rules, criteria) {
+          if (criteria.length > 0) {
+            _.forEach(group_by_criteria(rules, criteria[0]), function(rule) {
+              last_rules = rule;
+              return recurse(rule, criteria.slice(1));
+            });
+          } else {
+            if (last_rules.length > 0) $scope.is_duplicate = true;
+          }
+        };
+
+        var criteria = ['table_name', 'condition', 'field', 'data_type', 'min', 'max', 'units', 'severity', 'label'];
+        recurse(rules, criteria);
+        if ($scope.is_duplicate) return Notification.error({message: "Duplicate rules detected.", delay: 10000});
```

Comment:
I'm having a bit of a harder time tracking the logic through this, but from what I can tell, each value for each condition triggers their own recursive step at each recursive step. This seems super inefficient, and is a bit confusing. 

As much as I appreciate recursion, I think we may be able to accomplish this without it. 

What if we use the `_.group_by` and group by concatenated values of each of the criteria you listed? I'm imagining group_by returning something like:
```
`{rule['table_name']}-{rule['condition']}-`
```
Would that work? Let me know what you think.

Reply:
My initial thought was to break down all the rules by each criteria and at last, when all criteria has been iterated, rules set is not 0 in length (the unique-ness of each rule should lead to a empty rule group), then it's determined that there exist duplicate rules. With using recursion here, and if in the future more criteria is requested, we will just need to put the addition criteria in the criteria array.

Reply:
> ... if in the future more criteria is requested, we will just need to put the addition criteria in the criteria array.

I was thinking that grouping by concatenated values would address this as well. I'm still not convinced we need to have a recursive solution unless there's something I'm missing about the group by concatenated values approach, but let's chat about via call.